pkg/providers/cloudstack/cloudstack_test.go

```diff
@@ -250,6 +251,32 @@ func TestProviderGenerateDeploymentFileWithMirrorAndCertConfig(t *testing.T) {
 	test.AssertContentToFile(t, string(md), "testdata/expected_results_mirror_config_with_cert_md.yaml")
 }
 
+func TestProviderGenerateDeploymentFileWithProxyConfig(t *testing.T) {
+	clusterSpecManifest := "cluster_minimal_proxy.yaml"
+	mockCtrl := gomock.NewController(t)
+	setupContext()
+	kubectl := mocks.NewMockProviderKubectlClient(mockCtrl)
+	cluster := &types.Cluster{Name: "test"}
+	clusterSpec := givenClusterSpec(t, clusterSpecManifest)
+	datacenterConfig := givenDatacenterConfig(t, clusterSpecManifest)
+	machineConfigs := givenMachineConfigs(t, clusterSpecManifest)
+	ctx := context.Background()
+	cmk := givenWildcardCmk(mockCtrl)
+	provider := newProviderWithKubectl(t, datacenterConfig, machineConfigs, clusterSpec.Cluster, kubectl, cmk)
+
+	if err := provider.SetupAndValidateCreateCluster(ctx, clusterSpec); err != nil {
```

Comment:
Offline discussion - better to test GenerateCapiSpecForCreate independently, not rely on the logic in SetupAndValidateCreateCluster

Reply:
We need to call both because we're asserting on a legitimate and valid output file. `SetupAndValidateCreateCluster` sets some internal provider attributes which are then used by `GenerateCAPISpecForCreate`. If we want to isolate a test to only `GenerateCAPISpecForCreate`, this requires either manually modifying the internal components of the provider as part of the test setup, or asserting against an illogical/invalid output file.

I am inclined to keep the tests as they are for now, since this appears to be a deeper issue than malformed tests. There is an implicit dependency between these two methods and I don't feel like this is the PR to unravel it. If you'd like, I'd be happy to create a GitHub issue to track it

Reply:
Got it. Yes, please create a ticket. Having an implicit dependency between two exported methods is quite bad.

Reply:
Created https://github.com/aws/eks-anywhere/issues/1759. Please feel free to add additional details